### PR TITLE
Remove old selected_filters route 

### LINKED ISSUE #23632: Wrong rewrite url on category page
#### Describe the bug

If I add my text to the link (for example, /test) on the category page, error 404 should appear, but instead the link is overwritten and the product filter is included. nothing is happening on the page at this moment.

#### Expected behavior

Explain what you expected to happen instead.

#### Steps to Reproduce

1. Go to category page and add your text with slash for example /test
2. You can see url rewrite to ?selected_filters=test 
I believe that in this case, error 404 should appear. It is important.

**Screenshots**
First step https://drive.google.com/file/d/1RZG0FIJwg2AO1RuGmLp_CA-y_S3y6QMF/view?usp=sharing
Result https://drive.google.com/file/d/185Nkz7Zt942jXt2zqjvx3BrYJmu2Yi9Y/view?usp=sharing

#### Additional information

* PrestaShop version: 1.7.7.1
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by jf-viguier**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Remove selected_filters in routes : this route was created for [blocklayered](https://github.com/prestashop/blocklayered) module and this module was compatible with 1.6 and less only. This route cause a bug if you add a `/anything` at the end of the category url (cf issue)
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/23632
| How to test?      | Add `/anything` to a category url in front, it should display a not found page as expected (see issue). Please verify [ps_facetedsearch](https://github.com/PrestaShop/ps_facetedsearch) works well with this PR.
| Possible impacts? | none

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24509)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

**Comment by PrestaEdit** [comment]
We could also remove the else in Link class:
```
        else {
            $rule = 'layered_rule';
            $params['selected_filters'] = $selectedFilters;
        }
```

**Comment by PrestaEdit** [comment]
Also the ->add('layered_rule', TextType::class) into UrlSchemaType file.

**Review by matks — DISMISSED** [review]
I love PRs that remove code ❤️

**Comment by matks** [comment]
@jf-viguier I enhanced a bit your PR description to provide links and context to https://github.com/PrestaShop/ps_facetedsearch and https://github.com/prestashop/blocklayered. These links could help QA team to validate this PR has no issues.

**Comment by jf-viguier** [comment]
Thanks @PrestaEdit, I've removed old stuff in link class

File: classes/Link.php [hunk 1]
```diff
@@ -452,12 +452,7 @@ public function getCategoryLink(
         // Selected filters is used by the module ps_facetedsearch
         $selectedFilters = null === $selectedFilters ? '' : $selectedFilters;
```
Comment (Progi1984) [inline classes/Link.php]: Unuseful variables

**Review by Progi1984 — COMMENTED** [review]
Small feedbacks

File: classes/Link.php (hunk 1 above)
Comment (Progi1984) [reply classes/Link.php]:
```suggestion
```

Comment (jf-viguier) [reply classes/Link.php]: thanks, you means useless ?

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

File: classes/Link.php (hunk 1 above)
Comment (jf-viguier) [reply classes/Link.php]: line removed

**Comment by florine2623** [comment]
Hello @jf-viguier ,

Thanks for your PR.
Although, I'm having a little visual issue.

In PS 1.7.8 :
Screenshot 2021-06-15 at 10 20 06

In the PR :
Screenshot 2021-06-15 at 10 23 49

Could you check and feedback ?
Thanks !

**Comment by jf-viguier** [comment]
@florine2623 this little visual issue is not related to this PR but to the url structure with a new sub directory. You can open another issue.

**Comment by atomiix** [comment]
@florine2623 @jf-viguier The visual issue has been solved in the 1.7.8.x branch (https://github.com/PrestaShop/PrestaShop/pull/24852) once it will be merge into develop, it should be ok too ;)

**Comment by florine2623** [comment]
@jf-viguier @atomiix 

LGTM

Thanks!

**Comment by Progi1984** [comment]
Thanks @jf-viguier & @florine2623

**Pull request merged**